CONTRIBUTING_A_NEW_QUEST.md

```diff
@@ -0,0 +1,497 @@
+
+Change its class name and the file name to the new one.
+
+In copied code change package info (things like `package de.westnordost.streetcomplete.quests.defibrillator` at the top) to match the new folder containing the quest.
+
+# Add the quest to the list of active ones
+
+Adjust [QuestsModule.kt](app/src/main/java/de/westnordost/streetcomplete/quests/QuestsModule.kt) file. It contains a big list of active quests, ordered by priority. Read [what governs their priority](https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/quests/QuestsModule.kt#L169-L192) but do not worry too much, it can be tweaked later.
+
+Add your quest to the list so that it will be loaded by the app.
+
+## Quest anatomy
+See for example [simple yes/no quest asking whether AED is indoor or outdoor](app/src/main/java/de/westnordost/streetcomplete/quests/defibrillator/AddIsDefibrillatorIndoor.kt).
+
+
+TODO: maybe linking [app/src/main/java/de/westnordost/streetcomplete/data/osm/osmquests/OsmElementQuestType.kt](app/src/main/java/de/westnordost/streetcomplete/data/osm/osmquests/OsmElementQuestType.kt) would be better? And extend documentation there?
+
+### Element selection
+elementFilter property defines nodes, ways and relations which will be selected for a given quest. It is an element selection used by OsmFilterQuestType.
+
+```
+"""
+        nodes with
+         emergency = defibrillator
+         and access !~ private|no
+         and !indoor
+"""
+``` 
+
+This query will be limited to nodes (`nodes with`), which fulfill some requirements.
+
+- `emergency = defibrillator` tag must be present
+- `access` tag must not have values `private` or `no` to skip ones where mapper will be unable to survey (`!~ private|no` - to be more specific `optionA|optionB` is treated like `^optionA|optionB$` regexp)
+- `indoor` key must not be present at all, to show only ones where this tag is still missing
+
+It is specified as a string, in syntax specific to StreetComplete. You can look around some quests to see how it works. If you are trying to implement a new quest and you got stuck here, [open a new issue](https://github.com/streetcomplete/StreetComplete/issues) to request more thorough documentation here.
+
+###### Resurvey
+
+Some quests are asked not only when tag is missing but also when it is likely to be outdated.
+
+Typical code is in [quest asking about motorcycle parking capacity](app/src/main/java/de/westnordost/streetcomplete/quests/motorcycle_parking_capacity/AddMotorcycleParkingCapacity.kt):
+
+```
+    override val elementFilter = """
+        nodes, ways with amenity = motorcycle_parking
+         and access !~ private|no
+         and (!capacity or capacity older today -4 years)
+    """
+```  
+  
+This quest will be triggered when:
+
+ - on nodes and ways
+ - where `amenity = motorcycle_parking` tag is present
+ - `access` tag is not set with value `private` or `no`
```

Comment:
```suggestion
 - `access` tag does not have value `private` nor `no`
```

("is not set" sounds confusing to me here, as it can be read apart from the rest of the sentence, implying `access` is undefined). But I suspect even nicer English sentence is possible